packages/game-passes/test/GamePasses.test.ts

```diff
@@ -0,0 +1,1251 @@
+    });
+
+    it('should allow whitelisted addresses to transfer non-transferable tokens', async function () {
+      // Whitelist user1 for transfers
+      await sandboxPasses
+        .connect(admin)
+        .updateTransferWhitelist(TOKEN_ID_2, [user1.address], true);
+
+      // Now the transfer should succeed
+      await sandboxPasses
+        .connect(user1)
+        .safeTransferFrom(user1.address, user2.address, TOKEN_ID_2, 2, '0x');
+
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_2)).to.equal(
+        MINT_AMOUNT - 2,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_2)).to.equal(
+        2,
+      );
+    });
+
+    it('should allow admin to transfer non-transferable tokens', async function () {
+      // Approve admin to transfer
+      await sandboxPasses.connect(user1).setApprovalForAll(admin.address, true);
+
+      // Admin should be able to transfer non-transferable tokens
+      await sandboxPasses
+        .connect(admin)
+        .safeTransferFrom(user1.address, user2.address, TOKEN_ID_2, 2, '0x');
+
+      expect(await sandboxPasses.balanceOf(user1.address, TOKEN_ID_2)).to.equal(
+        MINT_AMOUNT - 2,
+      );
+      expect(await sandboxPasses.balanceOf(user2.address, TOKEN_ID_2)).to.equal(
+        2,
+      );
+    });
+  });
+
+  describe('Royalties', function () {
+    it('should return correct royalty info for default royalty', async function () {
+      const salePrice = ethers.parseEther('1');
+      const royaltyInfo = await sandboxPasses.royaltyInfo(
+        TOKEN_ID_1,
+        salePrice,
+      );
+
+      expect(royaltyInfo[0]).to.equal(royaltyReceiver.address);
+      expect(royaltyInfo[1]).to.equal(
+        (salePrice * ROYALTY_PERCENTAGE) / 10000n,
+      );
+    });
+
+    it('should allow admin to set default royalty', async function () {
+      const newRoyaltyPercentage = 1000n; // 10%
+
+      await sandboxPasses
+        .connect(admin)
+        .setDefaultRoyalty(user2.address, newRoyaltyPercentage);
+
+      const salePrice = ethers.parseEther('1');
+      const royaltyInfo = await sandboxPasses.royaltyInfo(
+        TOKEN_ID_1,
+        salePrice,
+      );
+
+      expect(royaltyInfo[0]).to.equal(user2.address);
+      expect(royaltyInfo[1]).to.equal(
+        (salePrice * newRoyaltyPercentage) / 10000n,
+      );
+    });
+
+    it('should allow admin to set token-specific royalty', async function () {
+      const tokenRoyaltyPercentage = 800n; // 8%
+
+      await sandboxPasses
+        .connect(admin)
+        .setTokenRoyalty(TOKEN_ID_2, user2.address, tokenRoyaltyPercentage);
+
+      const salePrice = ethers.parseEther('1');
+      const royaltyInfo = await sandboxPasses.royaltyInfo(
+        TOKEN_ID_2,
+        salePrice,
+      );
+
+      expect(royaltyInfo[0]).to.equal(user2.address);
+      expect(royaltyInfo[1]).to.equal(
+        (salePrice * tokenRoyaltyPercentage) / 10000n,
+      );
+    });
```

Comment:
missing tests for the above: cannot do these things if not admin

Reply:
Resolved in an earlier comment, all admin functions are now test for non-admin.